cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java

```diff
@@ -558,7 +560,8 @@ public void run(DatasetContext context) throws Exception {
   }
 
   @Override
-  public void setServiceInstances(final Id.Program id, final int instances) {
+  public void setServiceInstances(final ProgramId programId, final int instances) {
+    final Id.Program id = programId.toId();
```

Comment:
let's not do this. convert all private methods to use the new ``EntityId`` classes instead. Otherwise we will need one more PR for this class.